crates/red_knot_python_semantic/mdtest.py

```diff
@@ -125,22 +125,27 @@ def _run_mdtests_for_file(self, markdown_file: Path) -> None:
                 f"Test for [bold red]{markdown_file}[/bold red] failed",
                 style="gray",
             )
-            # Skip 'cargo test' boilerplate at the beginning:
-            lines = output.stdout.splitlines()
-            start_index = 0
-            for i, line in enumerate(lines):
-                if f"{test_name} stdout" in line:
-                    start_index = i
-                    break
-
-            for line in lines[start_index + 1 :]:
-                if "MDTEST_TEST_FILTER" in line:
-                    continue
-                if line.strip() == "-" * 50:
-                    # Skip 'cargo test' boilerplate at the end
-                    break
+            self._print_trimmed_cargo_test_output(
+                output.stdout + output.stderr, test_name
```

Comment:
`output.stdout + output.stderr` is the only relevant change. I just moved some code into a separate function.